### PR TITLE
feat: add coprime base

### PR DESCRIPTION
**Pull request by thofma**

### CODE REVIEW

**Comment by thofma** [comment]
Necessary for moving `factor` around. @fieker I renamed `coprime_base_insert` to `coprime_base_push!`.

File: src/algorithms/coprime_base.jl [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  T = Array{E}(undef, 1)
+  T[1] = S[1]
+  for i=2:length(S)
+    augment_coprime_base(T, S[i])
+  end
+  return T
+end
+
+@doc raw"""
+    coprime_base_insert(S::Vector{RingElem}, a::RingElem) -> Vector{RingElem}
```
Comment (thofma) [inline src/algorithms/coprime_base.jl:63]:
```suggestion
    coprime_base_push!(S::Vector{RingElem}, a::RingElem) -> Vector{RingElem}
```

Comment (lgoettgens) [inline src/algorithms/coprime_base.jl:63]:
```suggestion
    coprime_base_push!(S::Vector{RingElem}, a::RingElem) -> Vector{RingElem}
```

File: src/algorithms/coprime_base.jl [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+  return S
+end
+
+@doc raw"""
+    coprime_base(S::Vector{RingElem}) -> Vector{RingElem}
+
+Returns a coprime base for $S$, i.e. the resulting array contains pairwise coprime objects that multiplicatively generate the same set as the input array.
+"""
+function coprime_base(S::Vector{E}) where E
```
Comment (thofma) [inline src/algorithms/coprime_base.jl:52]:
```suggestion
function coprime_base(S::Vector{E}) where {E <: RingElem}
```

File: src/algorithms/coprime_base.jl [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+  return S
+end
+
+@doc raw"""
+    coprime_base(S::Vector{RingElem}) -> Vector{RingElem}
+
+Returns a coprime base for $S$, i.e. the resulting array contains pairwise coprime objects that multiplicatively generate the same set as the input array.
+"""
+function coprime_base(S::Vector{E}) where {E <: RingElement}
```
Comment (lgoettgens) [inline src/algorithms/coprime_base.jl:52]: why is this `RingElement` and the docstring says `RingElem`? can you make this consistent

File: src/algorithms/coprime_base.jl [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+  end
+  if !is_unit(a)
+    push!(S, a)
+  end
+
+  return S
+end
+
+@doc raw"""
+    coprime_base(S::Vector{RingElem}) -> Vector{RingElem}
```
Comment (thofma) [inline src/algorithms/coprime_base.jl:48]:
```suggestion
    coprime_base(S::Vector{RingElement}) -> Vector{RingElement}
```

**Pull request merged**